flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecSink.java

```diff
@@ -104,6 +118,10 @@ public class StreamExecSink extends CommonExecSink implements StreamExecNode<Obj
     @JsonInclude(JsonInclude.Include.NON_DEFAULT)
     private final boolean upsertMaterialize;
 
+    @JsonProperty(FIELD_NAME_UPSERT_MATERIALIZE_STRATEGY)
+    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+    private final SinkUpsertMaterializeStrategy upsertMaterializeStrategy;
```

Comment:
I take it that this code doesn't support any migration between different strategies? There is even no way for changing `VALUE`->`ADAPTIVE`?